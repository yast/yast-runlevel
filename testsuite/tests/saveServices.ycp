/**
 * YaST2 test case for runlevel editor
 *
 * tests function saveServices
 */
{
    // testedfiles: RunlevelEd.ycp Service.ycp Testsuite.ycp

    include "testsuite.ycp";

    map READ = $[
	"init": $[
	    "scripts": $[
		"exists": true,
		"default_runlevel" : "5",
		"current_runlevel" : "5",
		],
	    ],
	"target" : $[
	    "stat" : $[	"isreg" : true ],
            "size" : 0,
	    ]
	];
    TESTSUITE_INIT ([READ, $[], $[]], nil);

    import "RunlevelEd";
    import "Progress";

    Progress::off ();


    map EXEC = $[
	"target": $[
	    "bash_output": $[ "exit": 0, "stdout": "", "stderr": "", ],
	    ],
	];

    TEST (``(RunlevelEd::Write ()), [READ, $[], EXEC] , 0);

    RunlevelEd::services = $[
        "a" : $[
            "dirty" : false,
            "start" : ["1", "2", "3", ],
            ],
        "b" : $[
            "dirty" : false,
            "start" : ["1", "2", "3", ],
            ],
        ];
    TEST (``(RunlevelEd::Write ()), [READ, $[], EXEC] , 0);

    RunlevelEd::services = $[
        "a" : $[
            "dirty" : true,
            "start" : ["1", "2", "3", ],
            ],
        "b" : $[
            "dirty" : true,
            "start" : ["1", "2", "3", ],
            ],
        ];
    TEST (``(RunlevelEd::Write ()), [READ, $[], EXEC] , 0);

    RunlevelEd::services = $[
        "a" : $[
            "dirty" : true,
            "start" : ["1", "2", "3", ],
            "defstart" : ["1", "2", "S", "B" ],
            ],
        "b" : $[
            "dirty" : true,
            "start" : ["1", "2", "3", ],
            "defstart" : ["1", "2", "S", "b" ],
            ],
        ];
    TEST (``(RunlevelEd::Write ()), [READ, $[], EXEC] , 0);

    TEST (``(RunlevelEd::Write ()), [READ, $[], EXEC] , 0);

    RunlevelEd::services = $[
        "a" : $[
            "dirty" : true,
            "start" : [],
            "defstart" : ["1", "2", "S", ],
            ],
        "b" : $[
            "dirty" : true,
            "start" : [],
            "defstart" : ["1", "2", "S", ],
            ],
        ];
    TEST (``(RunlevelEd::Write ()), [READ, $[], EXEC] , 0);

    RunlevelEd::services = $[
        "nfsserver" : $[
            "dirty" : false,
            "start" : ["1"],
            "defstart" : ["1", "2", "S", ],
	    "reqstart" : [ "portmap", ],
            ],
        "portmap" : $[
            "dirty" : false,
            "start" : [],
            "defstart" : ["1", "2", "S", ],
            ],
        ];
    TEST (``(RunlevelEd::Write ()), [READ, $[], EXEC] , 0);

    RunlevelEd::services = $[
        "nfsserver" : $[
            "dirty" : true,
            "start" : ["1"],
            "defstart" : ["1", "2", "S", ],
	    "reqstart" : [ "portmap", ],
            ],
        "portmap" : $[
            "dirty" : false,
            "start" : [],
            "defstart" : ["1", "2", "S", ],
            ],
        ];
    TEST (``(RunlevelEd::Write ()), [READ, $[], EXEC] , 0);
    Progress::on ();
}
