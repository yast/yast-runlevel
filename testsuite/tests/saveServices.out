Execute	.target.bash "sed --in-place 's/^\\(l4:4:wait:\\/etc\\/init.d\\/rc 4\\)/#\\1/' /etc/inittab" 0
Read	.init.scripts.current_runlevel "5"
Return	true
Execute	.target.bash "sed --in-place 's/^\\(l4:4:wait:\\/etc\\/init.d\\/rc 4\\)/#\\1/' /etc/inittab" 0
Return	true
Execute	.target.bash "sed --in-place 's/^\\(l4:4:wait:\\/etc\\/init.d\\/rc 4\\)/#\\1/' /etc/inittab" 0
Log	Cannot enable service a (just) in selected runlevels, enabling in all default ones
Execute	.target.bash "/bin/systemctl is-enabled a.service" 0
Log	Cannot enable service b (just) in selected runlevels, enabling in all default ones
Execute	.target.bash "/bin/systemctl is-enabled b.service" 0
Return	true
Execute	.target.bash "sed --in-place 's/^\\(l4:4:wait:\\/etc\\/init.d\\/rc 4\\)/#\\1/' /etc/inittab" 0
Log	Cannot enable service a (just) in selected runlevels, enabling in all default ones
Execute	.target.bash "/bin/systemctl is-enabled a.service" 0
Log	Cannot enable service b (just) in selected runlevels, enabling in all default ones
Execute	.target.bash "/bin/systemctl is-enabled b.service" 0
Return	true
Execute	.target.bash "sed --in-place 's/^\\(l4:4:wait:\\/etc\\/init.d\\/rc 4\\)/#\\1/' /etc/inittab" 0
Log	Cannot enable service a (just) in selected runlevels, enabling in all default ones
Execute	.target.bash "/bin/systemctl is-enabled a.service" 0
Log	Cannot enable service b (just) in selected runlevels, enabling in all default ones
Execute	.target.bash "/bin/systemctl is-enabled b.service" 0
Return	true
Execute	.target.bash "sed --in-place 's/^\\(l4:4:wait:\\/etc\\/init.d\\/rc 4\\)/#\\1/' /etc/inittab" 0
Execute	.target.bash_output "/bin/systemctl --force disable a.service" $["exit":0, "stderr":"", "stdout":""]
Execute	.target.bash_output "/bin/systemctl --force disable b.service" $["exit":0, "stderr":"", "stdout":""]
Return	true
Execute	.target.bash "sed --in-place 's/^\\(l4:4:wait:\\/etc\\/init.d\\/rc 4\\)/#\\1/' /etc/inittab" 0
Return	true
Execute	.target.bash "sed --in-place 's/^\\(l4:4:wait:\\/etc\\/init.d\\/rc 4\\)/#\\1/' /etc/inittab" 0
Log	Cannot enable service nfsserver (just) in selected runlevels, enabling in all default ones
Execute	.target.bash "/bin/systemctl is-enabled nfsserver.service" 0
Return	true
